### ISSUE TITLE
Prometheus: Add support for `namespace`

### ISSUE THREAD

**Issue by andygrunwald** [labstack/echo-contrib#93]
### Issue Description

The current Prometheus middleware accepts only a subsystem argument:

```
p := prometheus.NewPrometheus("<subsystem>", nil)
p.Use(e)
```

Namespaces usually are used to prefix the metric name like

```
<namespace>_<subsystem>_<metricname>
```

This is helpful to avoid name conflicts (by using your application/company name as a namespace).

Every Metric type (e.g., Gaugse), inherits [Opts](https://pkg.go.dev/github.com/prometheus/client_golang@v1.14.0/prometheus#Opts) which supports a `Namespace` attribute.

https://github.com/labstack/echo-contrib/issues/80 and https://github.com/labstack/echo-contrib/pull/92 can go in the right direction, tho, namespace is not added to it yet.

### Checklist

- [X] Dependencies installed
- [X] No typos
- [X] Searched existing issues and docs

### Expected behaviour

Being able to accept a namespace AND subsystem name.

### Actual behaviour

Only the subsystem is accepted as an argument: https://echo.labstack.com/middleware/prometheus/

### Implementation

The current method signature looks like:

```go
func NewPrometheus(subsystem string, skipper middleware.Skipper, customMetricsList ...[]*Metric) *Prometheus {
...
```

A few other Prometheus implementations include the namespace as the first character, like:

```go
func NewPrometheus(namespace, subsystem string, skipper middleware.Skipper, customMetricsList ...[]*Metric) *Prometheus {
```

This would be a breaking change but would follow the "order" of how namespace and subsystem is used.

### Workaround

A possible workaround would be to set `subsystem` to `namespace_subsystem`.
Imagine your namespace is `tesla` and your subsystem is `modelx`, you would pass `tesla_modelx`.

### Version/commit

* github.com/labstack/echo-contrib v0.14.1

**Comment by yabberyabber** [comment]
A Namespace field would be very easy to add to the `PromethuesConfig` interface described in #92 - although I'm weary of this particular PR falling prey to feature creep so I suggest we open a new PR to add this functionality that's based on #92 

I think the `NewPrometheus` (without config) API should be considered deprecated once the `NewPrometheusWithConfig` API exists. It will need to stay around until the next major version for backwards-compatibility, but the way it's written makes it difficult to use for anything except the most basic usecases.

**Comment by andygrunwald** [comment]
@yabberyabber Sounds good. And I do agree with the feature creep comment.
I consider the namespace + subsystem variables still as very important for every prometheus instance. Renaming of metrics can lead to side effects later.

Do you need any kind of help integrating Namespaces in #92 ? Or do you aim get to #92 + running a new PR that adds Namespaces once #92 is merged?
And what is missing to get #92 out of the draft state?

Happy to help, once I know what is expected :)

**Comment by aldas** [comment]
closing, new middleware was added with https://github.com/labstack/echo-contrib/pull/94


namespace can be configured like that:
```go
package main

import (
	"errors"
	"github.com/labstack/echo-contrib/echoprometheus"
	"github.com/labstack/echo/v4"
	"log"
	"net/http"
)

func main() {
	e := echo.New()

	// add middleware with custom namespace and subsystem
	e.Use(echoprometheus.NewMiddlewareWithConfig(echoprometheus.MiddlewareConfig{
		Namespace: "ns",
		Subsystem: "ss",
	}))
	e.GET("/metrics", echoprometheus.NewHandler()) // register route where metrics can be scraped

	if err := e.Start(":8080"); err != nil && !errors.Is(err, http.ErrServerClosed) {
		log.Fatal(err)
	}
}

```
